server/athenian/api/align/models.py

```diff
@@ -84,22 +84,46 @@ def __init__(
         self,
         id: int,
         name: str,
-        children: List[TeamTree],
-        members: List[int],
+        children: list[TeamTree],
+        members: list[int],
+        members_count: Optional[int] = None,
+        total_members: Optional[list[int]] = None,
+        total_teams_count: Optional[int] = None,
+        total_members_count: Optional[int] = None,
     ):
         """Init the TeamTree."""
         self._id = id
         self._name = name
-        self._members = members
         self._children = children
+        self._members = members
+
+        if members_count is None:
+            members_count = len(members)
+        self._members_count = members_count
+
+        if total_members is None:
+            total_members = sorted(
+                set(chain(members, *(child.total_members for child in children))),
+            )
+        self._total_members = total_members
+
+        if total_teams_count is None:
+            total_teams_count = sum(child.total_teams_count for child in children) + len(children)
+        self._total_teams_count = total_teams_count
 
-        self._members_count = len(members)
-        self._total_members = sorted(
-            set(chain(members, *(child.total_members for child in children))),
-        )
-        self._total_teams_count = sum(child.total_teams_count for child in children) + len(
-            children,
-        )
+        if total_members_count is None:
+            total_members_count = len(total_members)
+        self._total_members_count = total_members_count
+
+    def with_children(self, children: list[TeamTree]) -> TeamTree:
+        """Return a copy of the object with children property replaced.
+
+        Properties depending from `children` retain the original value of the object.
+
+        """
+        copy = cast(TeamTree, self.copy())
```

Comment:
I think it is possible to change `copy()` to return the correct type https://stackoverflow.com/questions/68082661/typing-hinting-in-parent-class-for-a-method-that-returns-object-of-any-child-cla

Otherwise, we are heading towards a parody on Java/C#.